core/src/main/java/com/linecorp/armeria/common/logging/RequestRequestLog.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+import javax.net.ssl.SSLSession;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.client.ClientBuilder;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.Scheme;
+import com.linecorp.armeria.common.SerializationFormat;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.VirtualHostBuilder;
+
+import io.netty.channel.Channel;
+
+/**
+ * {@link RequestLog} properties available when request processing has completed.
+ */
+public interface RequestRequestLog {
```

Comment:
I'm not sure I like this name :laughing:

Reply:
You found my biggest trouble point ;)